src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +530,230 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
+        autoscale_configuration.recurrence.schedule.append(condition)
+    elif not autoscale_configuration.recurrence:
+        autoscale_configuration.recurrence = AutoscaleRecurrence(
+            schedule=[condition]
+        )
+    if no_wait:
+        return sdk_no_wait(no_wait, client.update_auto_scale_configuration, resource_group_name, cluster_name,
+                           autoscale_configuration)
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, days=None, time=None,
+                               workernode_count=None, no_wait=False):
+    cluster = client.get(resource_group_name, cluster_name)
+    autoscale_configuration = _extract_and_validate_autoscale_configuration(cluster)
+    _validate_schedule_configuration(autoscale_configuration)
+
+    if index >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError(
+            'This cluster only has {} condition(s). Please specify a correct index which starts with 0.'.format(
+                len(autoscale_configuration.recurrence.schedule)))
+
+    # try to update schedule condition
+    if days:
+        autoscale_configuration.recurrence.schedule[index].days = days
+    if time:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.time = time
+    if workernode_count:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.min_instance_count = workernode_count
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.max_instance_count = workernode_count
+
+    if no_wait:
+        return sdk_no_wait(no_wait, client.update_auto_scale_configuration, resource_group_name, cluster_name,
+                           autoscale_configuration)
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def delete_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, no_wait=False):
+    cluster = client.get(resource_group_name, cluster_name)
+    autoscale_configuration = _extract_and_validate_autoscale_configuration(cluster)
+    _validate_schedule_configuration(autoscale_configuration)
+
+    # convert index from list to set to remove duplicate
+    unique_index = set(index)
+    if len(unique_index) != len(index):
+        raise CLIError('There are duplicate indices in the value of --index.')
+
+    if len(unique_index) >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError('Deleting all conditions is not allowed.'
+                       'If you want to disable autoscale please use `az hdinsight autoscale delete`')
+
+    conditions_count = len(autoscale_configuration.recurrence.schedule)
+
+    # convert set to list to sort
+    unique_index = list(unique_index)
+    unique_index.sort()
+
+    delete_count = 0
+    for i in unique_index:
```

Comment:
>for i in unique_index: [](start = 4, length = 22)

need to handle i is negative

Reply:
fixed.

---
In reply to: [467852146](https://github.com/Azure/azure-cli/pull/14692#discussion_r467852146) [](ancestors = 467852146)